src/zos-io.cc

```diff
@@ -700,6 +714,62 @@ int __disableautocvt(int fd) {
   return fcntl(fd, F_CONTROL_CVT, &req);
 }
 
+int __ccsid_new_file() {
+  char* encode_file_new = getenv("_ENCODE_FILE_NEW");
+
+  int ccsid = 819;
+
+  if (encode_file_new)
+    if (strcmp(encode_file_new, "IBM-1047") == 0) {
+      ccsid = 1047;
+    } else if (strcmp(encode_file_new, "BINARY") == 0) {
+      // Set the file descriptor to binary mode
+      ccsid = FT_BINARY;
+    }
+
+  return ccsid;
+}
+
+int __tag_new_file_filename(int fd, std::string filename) {
+
+ auto it = existing_files.find(filename);
```

Comment:
Are you able to explain the logic here? If it's a new file, why would it find it in the existing_files map?

Reply:
I'm not entirely sure how vim does it, but saving to an existing file seems to rename the existing file and open a new file under the old name. In this case it is a new file, but since it matches a previously opened file we want to keep the old filetags.